src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationOptions.cs

```diff
@@ -152,5 +153,13 @@ public string CookieName
         /// to the client. This can be used to mitigate potential problems with very large identities.
         /// </summary>
         public IAuthenticationSessionStore SessionStore { get; set; }
+
+        CookieAuthenticationOptions IOptions<CookieAuthenticationOptions>.Value
```

Comment:
So I don't disagree, its just right now I'm only adding the overload to UseCookie.  When/if the time comes that we add this overload to more places, we can revisit this again.  For now its opt in and we can make this better as part of the larger sweep.  I don't want to add the Options class just for Cookies, I had it in my local changes, but reverted it as it wasn't used anywhere except for cookies yet...